src/providers/live/live-provider.ts

```diff
@@ -1,3 +1,222 @@
 import {Provider} from '../provider';
+import {KalturaCuePointType, CuepointTypeMap} from '../../types';
+import {
+  PushNotificationPrivider,
+  PushNotificationEventTypes,
+  SlideViewChangeNotificationsEvent,
+  ThumbNotificationsEvent,
+  PublicNotificationsEvent,
+  NotificationsErrorEvent,
+  SlideViewChangePushNotificationData,
+  ThumbPushNotificationData
+} from './push-notifications-provider';
+import {makeAssetUrl} from '../utils';
+import Player = KalturaPlayerTypes.Player;
+import Logger = KalturaPlayerTypes.Logger;
+import EventManager = KalturaPlayerTypes.EventManager;
 
-export class LiveProvider extends Provider {}
+const ID3_TAG_WAIT_TIMEOUT = 500;
+const ID3_TAG_LABEL = 'id3';
+
+export class LiveProvider extends Provider {
+  private _pushNotification: PushNotificationPrivider;
+  private _thumbCuePoints: ThumbPushNotificationData[] = [];
+  private _slideViewChangeCuePoints: SlideViewChangePushNotificationData[] = [];
+
+  private _id3Timestamp: number | null = 0;
+  private _currentTime = 0;
+  private _currentTimeLive = 0;
+  private _seekDifference: number | null = 0;
+  private _id3TagWaitTimeouts: Record<string, ReturnType<typeof setTimeout>> = {};
+
+  constructor(player: Player, eventManager: EventManager, logger: Logger, types: CuepointTypeMap) {
+    super(player, eventManager, logger, types);
+    this._pushNotification = new PushNotificationPrivider(this._player, this._logger);
+    this._initNotification();
+    this._pushNotification.registerToPushServer(this._player.getMediaInfo().entryId, types, this._handleConnection, this._handleConnectionError);
+    this._addBindings();
+  }
+
+  private _onTimedMetadataLoaded = ({payload}: any): void => {
+    // TODO: handle dash format
+    const id3TagCues = payload.label === ID3_TAG_LABEL ? payload.cues : [];
+    if (id3TagCues.length) {
+      try {
+        const id3Timestamp = Math.ceil(JSON.parse(id3TagCues[id3TagCues.length - 1].value.data).timestamp / 1000);
+        this._logger.debug(`Received new id3 timestamp: ${id3Timestamp}`);
```

Comment:
this will spam the log. is it important info?

Reply:
removed